src/analysis/transcendental/liouville.lean

```diff
@@ -0,0 +1,408 @@
+  ... = ∑ i in f.support, f.coeff i * a^i * b^(f.nat_degree - i:ℤ)/b^f.nat_degree
+      : sum_congr rfl (λ _ _, by rw [add_comm _ ↑f.nat_degree, sub_eq_add_neg])
+  ... = ∑ i in f.support, f.coeff i * a^i * b^(f.nat_degree - i)/b^f.nat_degree
+      : sum_congr rfl (λ _ h, by { rw [←int.sub_nat_nat_eq_coe, int.sub_nat_nat_of_ge (le_nat_degree_of_mem_support h), int.of_nat_eq_coe], refl })
+  ... = (∑ i in f.support, f.coeff i * a^i * b^(f.nat_degree - i))/b^f.nat_degree : by rw [sum_div]
+  ... = 1/b^f.nat_degree * (∑ i in f.support, f.coeff i*a^i*b^(f.nat_degree - i)) : by ring,
+have cast_eq : ∑ i in f.support, (f.coeff i : ℝ) * a ^ i * b ^ (f.nat_degree - i)
+            = ↑(∑ i in f.support, f.coeff i * a ^ i * b ^ (f.nat_degree - i)), from
+begin
+  induction f.support using finset.induction_on with a s H ih,
+  { simp only [int.cast_zero, sum_empty] },
+  { rw [sum_insert H, sum_insert H, ih],
+    simp only [int.cast_pow, int.cast_add, int.cast_mul] }
+end,
+begin
+  rw [eq₁, cast_eq, abs_mul, abs_of_pos pos, le_mul_iff_one_le_right pos],
+  rw [eq₁, mul_ne_zero_iff, cast_eq] at not_root,
+  norm_cast at ⊢ not_root,
+  suffices : 0 < abs (∑ (i : ℕ) in f.support, f.coeff i * a ^ i * b ^ (f.nat_degree - i)),
+  { omega },
+  { rw abs_pos_iff,
+    exact not_root.2 }
+end
+
+lemma exists_pos_real_of_irrational_root (α : ℝ) (hα : irrational α) (f : polynomial ℤ)
+  (f_deg : 1 < f.nat_degree) (α_root : (f.map (algebra_map ℤ ℝ)).eval α = 0) :
+  ∃ A : ℝ, 0 < A ∧ ∀ a b : ℤ, 0 < b → (A / b ^ (f.nat_degree)) < abs(α - a / b)  :=
+have f_nonzero : f ≠ 0, from ne_zero_of_nat_degree_gt f_deg,
+have f_ℝ_nonzero : (f.map (algebra_map ℤ ℝ)) ≠ 0, from
+begin
+  intro rid, refine f_nonzero _,
+  ext m, rw polynomial.ext_iff at rid, specialize rid m,
+  simp only [coeff_map, int.cast_eq_zero, ring_hom.eq_int_cast, coeff_zero] at ⊢ rid,
+  exact rid
+end,
+have H : _, from is_compact.exists_forall_ge (@compact_Icc (α-1) (α+1))
+          begin rw set.nonempty, use α, rw set.mem_Icc, split; linarith end
+          (show continuous_on (λ x, abs ((f.map (algebra_map ℤ ℝ)).derivative.eval x)) _,
+          begin apply continuous.continuous_on, continuity, exact continuous_abs end),
+
+begin
+  classical, rw irrational_iff_ne_rational at hα,
+  obtain ⟨x_max, ⟨x_max_range, hM⟩⟩ := H,
+  set M := abs ((f.map (algebra_map ℤ ℝ)).derivative.eval x_max),
+  have M_nonzero : M ≠ 0,
+  { intro absurd, rw absurd at hM,
+    replace hM := nat_degree_eq_zero_of_derivative_eq_zero
+      ((f.map (algebra_map ℤ ℝ)).derivative.eq_zero_of_infinite_is_root
+        (infinite_mono (λ y hy, by simp only [mem_set_of_eq, is_root.def, abs_nonpos_iff.1 (hM y hy)])
+        (Icc.infinite (show α - 1 < α + 1, by linarith)))),
+    rw [nat_degree_map' (show function.injective (algebra_map ℤ ℝ), from int.cast_injective)] at hM,
+    linarith only [f_deg, hM] },
+  have M_pos : 0 < M := lt_of_le_of_ne (abs_nonneg _) M_nonzero.symm,
+  clear M_nonzero,
+
```

Comment:
The empty line suggests that you are starting a new "step" or "part" of the proof. Could you please write short comments on these empty lines that explain the steps?